pkg/ble/device/device.go

```diff
@@ -0,0 +1,287 @@
+		collectCycle := time.Duration(dev.Instance.Twins[i].PVisitor.CollectCycle)
+		// If the collect cycle is not set, set it to 1 second.
+		if collectCycle == 0 {
+			collectCycle = 1 * time.Second
+		}
+		uuid := ble.MustParse(twinData.BluetoothVisitorConfig.CharacteristicUUID)
+		if p, err := twinData.BluetoothClient.Client.DiscoverProfile(true); err == nil {
+			if twinData.FindedCharacteristic = p.Find(ble.NewCharacteristic(uuid)); twinData.FindedCharacteristic == nil {
+				klog.Errorf("can't find uuid %s", uuid.String())
+				continue
+			}
+			c := twinData.FindedCharacteristic.(*ble.Characteristic)
+			// If this Characteristic suports notifications and there's a CCCD
+			// Then subscribe to it, the notifications operation is different from reading operation, notifications will keep looping when connected
+			// so we can't use imer.Start() for notifications
+			if (c.Property&ble.CharNotify) != 0 && c.CCCD != nil {
+				wg.Add(1)
+				go func() {
+					if err := twinData.BluetoothClient.Client.Subscribe(c, false, twinData.notificationHandler()); err != nil {
+						klog.Error(err)
+						wg.Done()
+					}
+				}()
+			} else if (c.Property & ble.CharRead) != 0 { // // read data actively
+				timer := mappercommon.Timer{Function: twinData.Run, Duration: collectCycle, Times: 0}
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					timer.Start()
+				}()
+			}
+		}
+	}
+}
+
+// initData initialize the timer to get data.
+func initData(dev *globals.BluetoothDev) {
+	for i := 0; i < len(dev.Instance.Datas.Properties); i++ {
+		var visitorConfig configmap.BluetoothVisitorConfig
+		if err := json.Unmarshal([]byte(dev.Instance.Twins[i].PVisitor.VisitorConfig), &visitorConfig); err != nil {
+			klog.Error(err)
+			continue
+		}
+
+		twinData := TwinData{
+			BluetoothClient:        dev.BluetoothClient,
+			Name:                   dev.Instance.Datas.Properties[i].PropertyName,
+			Type:                   dev.Instance.Datas.Properties[i].Metadatas.Type,
+			BluetoothVisitorConfig: visitorConfig,
+			Topic:                  fmt.Sprintf(mappercommon.TopicDataUpdate, dev.Instance.ID)}
+		collectCycle := time.Duration(dev.Instance.Datas.Properties[i].PVisitor.CollectCycle)
+		// If the collect cycle is not set, set it to 1 second.
+		if collectCycle == 0 {
+			collectCycle = 1 * time.Second
+		}
+		uuid := ble.MustParse(twinData.BluetoothVisitorConfig.CharacteristicUUID)
+		if p, err := twinData.BluetoothClient.Client.DiscoverProfile(true); err == nil {
+			if u := p.Find(ble.NewCharacteristic(uuid)); u == nil {
+				klog.Errorf("can't find uuid %s", uuid.String())
+				continue
+			}
+			timer := mappercommon.Timer{Function: twinData.Run, Duration: collectCycle, Times: 0}
+			// If this Characteristic suports notifications and there's a CCCD
+			// Then subscribe to it, the notifications operation is different from reading operation, notifications will keep looping when connected
```

Comment:
Correct the capital and small letter.
imer->timer

Reply:
/lgtm, it had been fixed